CollisionOperator/scatteringKernels_func.f90

```diff
@@ -149,6 +152,89 @@ function targetVelocity_constXS(E,dir,A,kT,rand) result (V_t)
 
   end function targetVelocity_constXS
 
+  !!
+  !! Function that returns a sample of target velocity with DBRC
+  !! V_t is a vector. The velocity is scaled by a factor sqrt(Mn/2) where Mn is mass of a neutron
+  !! so that V_t*V_t=E_t with E_t being kinetic energy of a NEUTRON traveling with TARGET VELOCITY
+  !! (note that it is not a kinetic energy of the target).
+  !!
+  !!
+  function targetVelocity_DBRCXS(nuc, E, dir, A, kT, rand, tempMaj) result (V_t)
+    class(ceNeutronNuclide), intent(in)     :: nuc
+    real(defReal), intent(in)               :: E
+    real(defReal), dimension(3), intent(in) :: dir
+    real(defReal), intent(in)               :: A
+    real(defReal), intent(in)               :: kT
+    real(defReal), intent(in)               :: tempMaj
+    class(RNG), intent(inout)               :: rand
+    real(defReal),dimension(3)              :: V_t
+    real(defReal)                           :: alpha, mu, phi, P_acc, DBRC_acc
+    real(defReal)                           :: X, Y
+    real(defReal)                           :: r1, r2, r3, r4, r5
+    real(defreal)                           :: rel_v, rel_E, xs_rel_v
+
+    ! Calculate neutron Y = beta *V_n
+    ! beta = sqrt(A*Mn/2kT). Note velocity scaling by sqrt(Mn/2).
+    Y = sqrt(A * E / kT)
+
+    ! Calculate threshold factor alpha
+    ! In MCNP, alpha is p1
+    alpha = 2.0 / (Y * sqrt(PI) + 2.0)
+
+    rejectionLoop: do
+
+      ! Obtain random numbers
+      r1 = rand % get()
+      r2 = rand % get()
+      r3 = rand % get()
+
+      ! Sample X = beta * V_t
+      ! Uses helper functions below
+      if ( r1 > alpha ) then
+        X = sample_x2expx2(rand)
+      else
+        X = sample_x3expx2(rand)
+      end if
+
+      ! Sample polar angle of target velocity wrt. neutron direction
+      mu = 2.0 * r2 - 1.0;
+
+      ! Calculate relative velocity between neutron and target
+      rel_v = sqrt(Y * Y + X * X - 2.0 * X * Y * mu)
+
+      ! Calculate Acceptance Propability
+      P_acc = rel_v / (Y + X)
+
+      ! Accept or reject mu and target velocity
+      if (P_acc < r3) cycle rejectionLoop
+
+      ! Relative energy = relative velocity **2 due to sqrt(Mn/2) scaling factor
+      rel_E = (rel_v**2 * kT / A)
+
+      ! Find 0K scattering xs of target at relative energy
+      xs_rel_v = nuc % elScatteringXS(rel_E)
```

Comment:
Perhaps it would be a good idea to move the common part of the sampling between DBRC and non-DBRC target velocity sampling to a shared, separate procedure (private for the module)? It would avoid some code duplication.